tests/unit/test_dataset.py

```diff
@@ -150,8 +172,10 @@ def test_from_api_repr(self):
 
     def test_from_api_repr_w_unknown_entity_type(self):
```

Comment:
Could we also add a `from_api_repr` -> `to_api_repr` round-trip test with unknown entity type (and even additional properties that aren't the entity type)? Could be an extra set of assertions on this test or a separate test.

`to_api_repr` is the most important piece, as that's what determines what actually gets sent to the API.

Reply:
Updated